clicommand/agent_start.go

```diff
@@ -536,7 +536,11 @@ var AgentStartCommand = cli.Command{
 
 		// Set a useful default for the bootstrap script
 		if cfg.BootstrapScript == "" {
-			cfg.BootstrapScript = fmt.Sprintf("%s bootstrap", shellwords.Quote(os.Args[0]))
+			cfg.BootstrapScript = fmt.Sprintf(
+				"%s bootstrap --cancel-signal %s",
+				shellwords.Quote(os.Args[0]),
+				cfg.CancelSignal,
+			)
```

Comment:
This is the part that I think should be propagated via environment rather than CLI args. I think we have other options passing through via environment, perhaps most recently added by @ticky?